src/lib.nr

```diff
@@ -200,13 +223,132 @@ fn test_base64_decode_elements() {
         6, 49, 12, 37, 32, 48, 11, 34, 50, 41, 39, 21, 43, 4, 54, 2, 52, 18, 31, 56, 50, 51, 33, 28, 22, 19, 36, 0, 33, 18, 25, 57, 62, 22, 4, 17, 33, 10, 33, 23, 45, 37, 20
     ];
 
-    let ascii_result = base64_decode_elements(input);
+    let ascii_result = base64_encode_elements(input);
 
     assert(ascii_result == ascii_expected);
 }
 
 #[test]
-fn test_base64_encode() {
+fn test_encode_empty() {
+    let input: [u8; 0] = [];
+    let result = base64_encode(input);
+    let expected: [u8; 0] = [];
+
+    assert(result == expected);
+}
+
+#[test]
+fn test_decode_empty() {
+    let input: [u8; 0] = [];
+    let expected: [u8; 0] = [];
+    let result = base64_decode(input);
+    assert(result == expected);
+}
+
+#[test]
+fn test_encode_max_byte() {
+    let input: [u8; 1] = [255];
+
+    let result: [u8; 2] = base64_encode(input);
+    let expected: [u8; 2] = [47, 119]; // "/w"
+    assert(result == expected);
+}
+
+#[test]
+fn test_decode_max_byte() {
+    let expected: [u8; 1] = [255];
+
+    let input: [u8; 2] = [47, 119]; // "/w"
+    let result: [u8; 1] = base64_decode(input);
+    assert(result == expected);
+}
+
+#[test(should_fail_with="DecodeError: invalid symbol 255, offset 0")]
+fn test_decode_invalid() {
+    let input: [u8; 1] = [255];
+    let _: [u8; 0] = base64_decode(input);
+}
+
+#[test]
+fn test_encode_ascii() {
+    // "Hello World!"
+    let input: [u8; 12] = [72, 101, 108, 108, 111, 32, 87, 111, 114, 108, 100, 33];
+    // base64: SGVsbG8gV29ybGQh
+    let expected: [u8; 16] = [83, 71, 86, 115, 98, 71, 56, 103, 86, 50, 57, 121, 98, 71, 81, 104];
+
+    // all configurations should give the same encoding
+    let result = base64_encode(input);
+    assert(result == expected);
+}
+
+#[test]
+fn test_decode_ascii() {
+    // base64: SGVsbG8gV29ybGQh
+    let input: [u8; 16] = [83, 71, 86, 115, 98, 71, 56, 103, 86, 50, 57, 121, 98, 71, 81, 104];
+    // "Hello World!"
+    let expected: [u8; 12] = [72, 101, 108, 108, 111, 32, 87, 111, 114, 108, 100, 33];
+
+    // all configurations should decode the same way
+    let result: [u8; 12] = base64_decode(input);
+    assert(result == expected);
+}
+
+#[test]
+fn test_encode_utf8() {
+    // non-ascii utf-8: "Hello, World!" in Japanese: "こんにちは、世界！"
```

Comment:
Can you try with noir version 0.35.0 please. Non-ascii characters in comments are not supported in the latest version.

Reply:
Yep, this gives errors. I can remove the japanese from the comment